detection_rules/rule_validators.py

```diff
@@ -136,10 +146,22 @@ def validate(self, data: QueryRuleData, meta: RuleMeta) -> None:
                 validation_checks["integrations"] = self.validate_integration(data, meta, package_integrations)
 
             if (validation_checks["stack"] and not package_integrations):
-                raise validation_checks["stack"]
+                # if auto add, try auto adding and then call stack_combo validation again
+                if validation_checks["stack"].error_msg == "Unknown field" and RULES_CONFIG.auto_gen_schema_file:
+                    # auto add the field and re-validate
+                    self.auto_add_field(validation_checks["stack"], data.index_or_dataview[0])
+                    self.validate(data, meta, depth + 1)
+                else:
+                    raise validation_checks["stack"]
 
             if (validation_checks["stack"] and validation_checks["integrations"]):
-                raise ValueError(f"Error in both stack and integrations checks: {validation_checks}")
+                # if auto add, try auto adding and then call stack_combo validation again
+                if validation_checks["stack"].error_msg == "Unknown field" and RULES_CONFIG.auto_gen_schema_file:
+                    # auto add the field and re-validate
+                    self.auto_add_field(validation_checks["stack"], data.index_or_dataview[0])
+                    self.validate(data, meta, depth + 1)
+                else:
+                    raise ValueError(f"Error in both stack and integrations checks: {validation_checks}")
```

Comment:
nit: might be able to make two small methods since this looks very similar to the code in the other `validate` method.

Reply:
That is true, but I think it would make the recursion harder to follow. But, I might just be missing something as to how this would be split.

Reply:
Good point. FWIW, you may not need recursion at all. 

<details><summary>Details</summary>
<p>

```python
def validate(self, data: 'QueryRuleData', meta: RuleMeta, max_attempts: int = 10) -> None:
    """Validate an EQL query while checking TOMLRule."""
    if meta.query_schema_validation is False or meta.maturity == "deprecated":
        # syntax only, which is done via self.ast
        return

    if isinstance(data, QueryRuleData) and data.language != 'lucene':
        packages_manifest = load_integrations_manifests()
        package_integrations = TOMLRuleContents.get_packaged_integrations(data, meta, packages_manifest)

        for attempt in range(max_attempts):
            validation_checks = {"stack": None, "integrations": None}
            # validate the query against fields within beats
            validation_checks["stack"] = self.validate_stack_combos(data, meta)

            if package_integrations:
                # validate the query against related integration fields
                validation_checks["integrations"] = self.validate_integration(data, meta, package_integrations)

            if validation_checks["stack"] and not package_integrations:
                # if auto add, try auto adding and then validate again
                if "Field not recognized" in validation_checks["stack"].error_msg and RULES_CONFIG.auto_gen_schema_file:
                    # auto add the field and re-validate
                    self.auto_add_field(validation_checks["stack"], data.index_or_dataview[0])
                else:
                    raise validation_checks["stack"]

            elif validation_checks["stack"] and validation_checks["integrations"]:
                # if auto add, try auto adding and then validate again
                if "Field not recognized" in validation_checks["stack"].error_msg and RULES_CONFIG.auto_gen_schema_file:
                    # auto add the field and re-validate
                    self.auto_add_field(validation_checks["stack"], data.index_or_dataview[0])
                else:
                    raise ValueError(f"Error in both stack and integrations checks: {validation_checks}")

            else:
                break

        else:
            raise ValueError("Maximum validation attempts exceeded")

        rule_type_config_fields, rule_type_config_validation_failed = \
            self.validate_rule_type_configurations(data, meta)
        if rule_type_config_validation_failed:
            raise ValueError(f"""Rule type config values are not ECS compliant, check these values:
                            {rule_type_config_fields}""")
```

</p>
</details>

Reply:
Updated :+1: